core/src/main/scala/org/http4s/headers/Expires.scala

```diff
@@ -7,12 +7,29 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.Parser
+import cats.syntax.all._
 import org.http4s.util.{Renderer, Writer}
 
 object Expires extends HeaderKey.Internal[Expires] with HeaderKey.Singleton {
   override def parse(s: String): ParseResult[Expires] =
-    HttpHeaderParser.EXPIRES(s)
+    parser.parseAll(s).leftMap { e =>
+      ParseFailure("Invalid Expires header", e.toString)
+    }
+
+  /* `Expires = HTTP-date` */
+  private[http4s] val parser: Parser[Expires] = {
```

Comment:
As a consequence of treating invalid dates as special, this becomes a `Parser` instead of a `Parser1`.  I don't _think_ that will haunt us, but, maybe?